ckanext/activity/logic/action.py

```diff
@@ -257,7 +258,22 @@ def package_activity_list(
         exclude_activity_types=exclude_activity_types,
     )
 
-    return model_activity.activity_list_dictize(activity_objects, context)
+    # Filter based on user permissions
+    filtered_activity_objects = []
+    user = context.get('user')
+    if not authz.is_sysadmin(user):
+        labels = get_permission_labels()
+        user_permission_labels = labels.get_user_activity_labels(context['auth_user_obj'])
+
+        for activity in activity_objects:
+            activity.permission_labels = labels.get_activity_labels(activity)
+
+            if activity.permission_labels and any(permission in activity.permission_labels for permission in user_permission_labels):
+                filtered_activity_objects.append(activity)
```

Comment:
this should be done by the database not in python, or it's going to be very very slow